tests/UnitTests/Foundation/DispatchTests.mm

```diff
@@ -0,0 +1,200 @@
+    
+    dispatch_group_async(group, queue, ^{
+		EXPECT_EQ(dispatch_get_specific(key), context2);
+    });
+	ASSERT_EQ(dispatch_group_wait(group, _createEndTestTime()), 0);
+
+	dispatch_release(group);
+	dispatch_release(queue);
+}
+
+TEST(DispatchTests, MultipleSpecificKeys) {
+    dispatch_queue_t queue = dispatch_queue_create("queue", nullptr);
+	dispatch_group_t group = dispatch_group_create();
+
+    dispatch_queue_set_specific(queue, key, context, nullptr);
+	dispatch_queue_set_specific(queue, key2, context2, nullptr);
+    
+    dispatch_group_async(group, queue, ^{
+		EXPECT_EQ(dispatch_get_specific(key), context);
+		EXPECT_EQ(dispatch_get_specific(key2), context2);
+        dispatch_queue_set_specific(queue, key, context2, nullptr);
+		dispatch_queue_set_specific(queue, key2, context, nullptr);
+    });
+	ASSERT_EQ(dispatch_group_wait(group, _createEndTestTime()), 0);
+    
+	EXPECT_EQ(dispatch_queue_get_specific(queue, key), context2);
+	EXPECT_EQ(dispatch_queue_get_specific(queue, key2), context);
+
+	dispatch_queue_set_specific(queue, key3, context3, nullptr);
+    
+    dispatch_group_async(group, queue, ^{
+		EXPECT_EQ(dispatch_get_specific(key), context2);
+		EXPECT_EQ(dispatch_get_specific(key2), context);
+		EXPECT_EQ(dispatch_get_specific(key3), context3);
+    });
+	ASSERT_EQ(dispatch_group_wait(group, _createEndTestTime()), 0);
+
+	dispatch_release(queue);
+	dispatch_release(group);
+}
+
+TEST(DispatchTests, SpecificDestructor) {
+	dispatch_queue_t queue = dispatch_queue_create("queue", nullptr);
+	dispatch_group_t group = dispatch_group_create();
+
+    dispatch_queue_set_specific(queue, key, context, destructor);
+
+	EXPECT_EQ(destructorCount, 0);
+    
+    dispatch_group_async(group, queue, ^{
+		EXPECT_EQ(dispatch_get_specific(key), context);
+        dispatch_queue_set_specific(queue, key, context2, destructor);
+		EXPECT_EQ(destructorCount, 1);
+    });
+	ASSERT_EQ(dispatch_group_wait(group, _createEndTestTime()), 0);
+    
+	EXPECT_EQ(dispatch_queue_get_specific(queue, key), context2);
+	dispatch_queue_set_specific(queue, key, context, destructor);
+	EXPECT_EQ(destructorCount, 2);
+    
+    dispatch_group_async(group, queue, ^{
+		EXPECT_EQ(dispatch_get_specific(key), context);
+		EXPECT_EQ(destructorCount, 2);
+    });
+	ASSERT_EQ(dispatch_group_wait(group, _createEndTestTime()), 0);
+
+	dispatch_release(queue);
+	dispatch_release(group);
+}
+
+TEST(DispatchTests, RemoveSpecific) {
+	dispatch_queue_t queue = dispatch_queue_create("queue", nullptr);
+	dispatch_group_t group = dispatch_group_create();
+
+    dispatch_queue_set_specific(queue, key, context, nullptr);
+    
+    dispatch_group_async(group, queue, ^{
+		EXPECT_EQ(dispatch_get_specific(key), context);
+        dispatch_queue_set_specific(queue, key, nullptr, nullptr);
+    });
+	ASSERT_EQ(dispatch_group_wait(group, _createEndTestTime()), 0);
+
+	EXPECT_EQ(dispatch_queue_get_specific(queue, key), nullptr);
```

Comment:
Minor nit: The EXPECT and ASSERT macros work better if the “expected” value is on the left.

Reply:
Fixed.